### ISSUE TITLE
0.40.10 - worker fails with micronaut server error [${airbyte.local.docker-mount}]. Property doesn't exist - kubernetes manifest deployment

### ISSUE THREAD

**Issue by andnig** [airbytehq/airbyte#17595]
## Environment
- **Airbyte version**:  0.40.10
- **OS Version / Instance**: GCP container optimized linux
- **Deployment**: Kubernetes deployment with repositories kubernetes yaml files
- **Step where error happened**: Upgrade Airbyte

## Current Behavior
When upgrading airbyte from 0.40.9 to 0.40.10 using the kubernetes manifest files in this repository, the worker constantly crashes during startup with the following exception.
I'm using the kubernetes files as-is from this repository (``kubectl apply -k kube/overlays/stable``).
- The worker.yaml is unchanged
- The .env is unchanged
- The kustomization.yaml is unchanged - except the namespace was changed to "airbyte"

```
2022-10-05 06:56:52 ERROR i.m.r.Micronaut(handleStartupException):338 - Error starting Micronaut server: Failed to inject value for parameter [localDockerMount] of method [checkDockerProcessFactory] of class: io.airbyte.workers.process.ProcessFactory

Message: Failed to inject value for parameter [localDockerMount] of method [checkDockerProcessFactory] of class: io.airbyte.workers.process.ProcessFactory

Message: Error resolving property value [${airbyte.local.docker-mount}]. Property doesn't exist
Path Taken: new ApplicationInitializer() --> ApplicationInitializer.checkConnectionActivities --> List.checkConnectionActivities([CheckConnectionActivity checkConnectionActivity]) --> new CheckConnectionActivityImpl(WorkerConfigs workerConfigs,[ProcessFactory processFactory],SecretsHydrator secretsHydrator,Path workspaceRoot,WorkerEnvironment workerEnvironment,LogConfigs logConfigs,AirbyteApiClient airbyteApiClient,String airbyteVersion) --> ProcessFactory.checkDockerProcessFactory(WorkerConfigs workerConfigs,String dockerNetwork,[String localDockerMount],String localRoot,String workspaceDockerMount,String workspaceRoot)
Path Taken: new ApplicationInitializer() --> ApplicationInitializer.checkConnectionActivities --> List.checkConnectionActivities([CheckConnectionActivity checkConnectionActivity]) --> new CheckConnectionActivityImpl(WorkerConfigs workerConfigs,[ProcessFactory processFactory],SecretsHydrator secretsHydrator,Path workspaceRoot,WorkerEnvironment workerEnvironment,LogConfigs logConfigs,AirbyteApiClient airbyteApiClient,String airbyteVersion) --> ProcessFactory.checkDockerProcessFactory(WorkerConfigs workerConfigs,String dockerNetwork,[String localDockerMount],String localRoot,String workspaceDockerMount,String workspaceRoot)
io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [localDockerMount] of method [checkDockerProcessFactory] of class: io.airbyte.workers.process.ProcessFactory
```

## Expected Behavior
Worker should start up with v 0.40.10

## Logs
```

    ___    _      __          __
   /   |  (_)____/ /_  __  __/ /____
  / /| | / / ___/ __ \/ / / / __/ _ \
 / ___ |/ / /  / /_/ / /_/ / /_/  __/
/_/  |_/_/_/  /_.___/\__, /\__/\___/
                    /____/
        : airbyte-workers :

  Micronaut (v3.6.3)

2022-10-05 06:56:51 INFO i.m.c.e.DefaultEnvironment(<init>):159 - Established active environments: [k8s, cloud, gcp, control-plane]
2022-10-05 06:56:51 INFO c.z.h.HikariDataSource(<init>):71 - HikariPool-1 - Starting...
2022-10-05 06:56:51 INFO c.z.h.HikariDataSource(<init>):73 - HikariPool-1 - Start completed.
2022-10-05 06:56:51 INFO c.z.h.HikariDataSource(<init>):71 - HikariPool-2 - Starting...
2022-10-05 06:56:51 INFO c.z.h.HikariDataSource(<init>):73 - HikariPool-2 - Start completed.
2022-10-05 06:56:51 INFO i.m.l.PropertiesLoggingLevelsConfigurer(configureLogLevelForPrefix):107 - Setting log level 'DEBUG' for logger: 'io.airbyte.bootloader'
2022-10-05 06:56:52 ERROR i.m.r.Micronaut(handleStartupException):338 - Error starting Micronaut server: Failed to inject value for parameter [localDockerMount] of method [checkDockerProcessFactory] of class: io.airbyte.workers.process.ProcessFactory

Message: Failed to inject value for parameter [localDockerMount] of method [checkDockerProcessFactory] of class: io.airbyte.workers.process.ProcessFactory

Message: Error resolving property value [${airbyte.local.docker-mount}]. Property doesn't exist
Path Taken: new ApplicationInitializer() --> ApplicationInitializer.checkConnectionActivities --> List.checkConnectionActivities([CheckConnectionActivity checkConnectionActivity]) --> new CheckConnectionActivityImpl(WorkerConfigs workerConfigs,[ProcessFactory processFactory],SecretsHydrator secretsHydrator,Path workspaceRoot,WorkerEnvironment workerEnvironment,LogConfigs logConfigs,AirbyteApiClient airbyteApiClient,String airbyteVersion) --> ProcessFactory.checkDockerProcessFactory(WorkerConfigs workerConfigs,String dockerNetwork,[String localDockerMount],String localRoot,String workspaceDockerMount,String workspaceRoot)
Path Taken: new ApplicationInitializer() --> ApplicationInitializer.checkConnectionActivities --> List.checkConnectionActivities([CheckConnectionActivity checkConnectionActivity]) --> new CheckConnectionActivityImpl(WorkerConfigs workerConfigs,[ProcessFactory processFactory],SecretsHydrator secretsHydrator,Path workspaceRoot,WorkerEnvironment workerEnvironment,LogConfigs logConfigs,AirbyteApiClient airbyteApiClient,String airbyteVersion) --> ProcessFactory.checkDockerProcessFactory(WorkerConfigs workerConfigs,String dockerNetwork,[String localDockerMount],String localRoot,String workspaceDockerMount,String workspaceRoot)
io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [localDockerMount] of method [checkDockerProcessFactory] of class: io.airbyte.workers.process.ProcessFactory

Message: Failed to inject value for parameter [localDockerMount] of method [checkDockerProcessFactory] of class: io.airbyte.workers.process.ProcessFactory

Message: Error resolving property value [${airbyte.local.docker-mount}]. Property doesn't exist
Path Taken: new ApplicationInitializer() --> ApplicationInitializer.checkConnectionActivities --> List.checkConnectionActivities([CheckConnectionActivity checkConnectionActivity]) --> new CheckConnectionActivityImpl(WorkerConfigs workerConfigs,[ProcessFactory processFactory],SecretsHydrator secretsHydrator,Path workspaceRoot,WorkerEnvironment workerEnvironment,LogConfigs logConfigs,AirbyteApiClient airbyteApiClient,String airbyteVersion) --> ProcessFactory.checkDockerProcessFactory(WorkerConfigs workerConfigs,String dockerNetwork,[String localDockerMount],String localRoot,String workspaceDockerMount,String workspaceRoot)
Path Taken: new ApplicationInitializer() --> ApplicationInitializer.checkConnectionActivities --> List.checkConnectionActivities([CheckConnectionActivity checkConnectionActivity]) --> new CheckConnectionActivityImpl(WorkerConfigs workerConfigs,[ProcessFactory processFactory],SecretsHydrator secretsHydrator,Path workspaceRoot,WorkerEnvironment workerEnvironment,LogConfigs logConfigs,AirbyteApiClient airbyteApiClient,String airbyteVersion) --> ProcessFactory.checkDockerProcessFactory(WorkerConfigs workerConfigs,String dockerNetwork,[String localDockerMount],String localRoot,String workspaceDockerMount,String workspaceRoot)
        at io.micronaut.context.exceptions.DependencyInjectionException.missingProperty(DependencyInjectionException.java:289) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractInitializableBeanDefinition.lambda$resolvePropertyValue$10(AbstractInitializableBeanDefinition.java:2043) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
        at io.micronaut.context.AbstractInitializableBeanDefinition.resolvePropertyValue(AbstractInitializableBeanDefinition.java:2043) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractInitializableBeanDefinition.getPropertyPlaceholderValueForConstructorArgument(AbstractInitializableBeanDefinition.java:1402) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.airbyte.workers.config.$ProcessFactoryBeanFactory$CheckDockerProcessFactory0$Definition.build(Unknown Source) ~[io.airbyte-airbyte-workers-0.40.10.jar:?]
        at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2354) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2305) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2251) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3016) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2918) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2879) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2800) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1617) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:66) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2065) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConstructorArgument(AbstractInitializableBeanDefinition.java:1297) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.airbyte.workers.temporal.check.connection.$CheckConnectionActivityImpl$Definition.build(Unknown Source) ~[io.airbyte-airbyte-workers-0.40.10.jar:?]
        at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2354) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2305) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2251) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3016) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2918) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2879) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2800) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1617) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:66) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2065) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConstructorArgument(AbstractInitializableBeanDefinition.java:1297) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.airbyte.workers.config.$ActivityBeanFactory$CheckConnectionActivities0$Definition.build(Unknown Source) ~[io.airbyte-airbyte-workers-0.40.10.jar:?]
        at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2354) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2305) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2251) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3016) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2918) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2879) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2800) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.findBean(DefaultBeanContext.java:1680) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractBeanResolutionContext.findBean(AbstractBeanResolutionContext.java:84) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractInitializableBeanDefinition.resolveOptionalBean(AbstractInitializableBeanDefinition.java:2178) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.AbstractInitializableBeanDefinition.findBeanForField(AbstractInitializableBeanDefinition.java:1843) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.airbyte.workers.$ApplicationInitializer$Definition.injectBean(Unknown Source) ~[io.airbyte-airbyte-workers-0.40.10.jar:?]
        at io.airbyte.workers.$ApplicationInitializer$Definition.build(Unknown Source) ~[io.airbyte-airbyte-workers-0.40.10.jar:?]
        at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2354) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2305) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2251) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3016) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2918) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2879) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2853) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.addCandidateToList(DefaultBeanContext.java:3511) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.resolveBeanRegistrations(DefaultBeanContext.java:3457) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.getBeanRegistrations(DefaultBeanContext.java:3427) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.getBeansOfType(DefaultBeanContext.java:1381) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.getBeansOfType(DefaultBeanContext.java:903) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.DefaultBeanContext.getBeansOfType(DefaultBeanContext.java:893) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.context.event.ApplicationEventPublisherFactory$2.lambda$$0(ApplicationEventPublisherFactory.java:217) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.core.util.SupplierUtil$2.initialize(SupplierUtil.java:77) ~[micronaut-core-3.6.3.jar:3.6.3]
        at io.micronaut.core.util.SupplierUtil$2.get(SupplierUtil.java:72) ~[micronaut-core-3.6.3.jar:3.6.3]
        at io.micronaut.context.event.ApplicationEventPublisherFactory$2.publishEvent(ApplicationEventPublisherFactory.java:229) ~[micronaut-inject-3.6.3.jar:3.6.3]
        at io.micronaut.http.server.netty.NettyHttpServer.lambda$fireStartupEvents$15(NettyHttpServer.java:574) ~[micronaut-http-server-netty-3.6.3.jar:3.6.3]
        at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
        at io.micronaut.http.server.netty.NettyHttpServer.fireStartupEvents(NettyHttpServer.java:568) ~[micronaut-http-server-netty-3.6.3.jar:3.6.3]
        at io.micronaut.http.server.netty.NettyHttpServer.start(NettyHttpServer.java:297) ~[micronaut-http-server-netty-3.6.3.jar:3.6.3]
        at io.micronaut.http.server.netty.NettyHttpServer.start(NettyHttpServer.java:104) ~[micronaut-http-server-netty-3.6.3.jar:3.6.3]
        at io.micronaut.runtime.Micronaut.lambda$start$2(Micronaut.java:81) ~[micronaut-context-3.6.3.jar:3.6.3]
        at java.util.Optional.ifPresent(Optional.java:178) ~[?:?]
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:79) ~[micronaut-context-3.6.3.jar:3.6.3]
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:323) ~[micronaut-context-3.6.3.jar:3.6.3]
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:309) ~[micronaut-context-3.6.3.jar:3.6.3]
        at io.airbyte.workers.Application.main(Application.java:12) ~[io.airbyte-airbyte-workers-0.40.10.jar:?]
```

## Steps to Reproduce
1. Use the kubernetes manifests in this repository
2. Run ``kubectl apply -k kube/overlays/stable``
--> worker fails to start

## Are you willing to submit a PR?
If you help me pinpoint the area of the issue, I am

**Comment by andnig** [comment]
Ok I got it running by manually adding the follgowing environment block to the worker.yaml
```yaml
            - name: WORKER_ENVIRONMENT
              valueFrom:
                configMapKeyRef:
                  name: airbyte-env
                  key: WORKER_ENVIRONMENT
```
Was this maybe deleted by accident?

**Comment by sh4sh** [comment]
Looks like it was removed in https://github.com/airbytehq/airbyte/pull/17388

**Comment by sh4sh** [comment]
Reproduced this issue with `kube/overlays/stable` and confirmed Helm is not affected

**Comment by navi86** [comment]
@sh4sh 
helm chart is affected as well, because changes were made here
https://github.com/airbytehq/airbyte/blob/master/charts/airbyte-worker/templates/deployment.yaml

**Comment by andnig** [comment]
The issue was resolved by updating to 0.40.14. Closing for that matter.
